File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.js
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     constructor(val = 0, next = null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

class Solution {
    /**
     * @param {ListNode} head
     * @return {ListNode}
     */
    reverseList(head) {

        let prev = null;
        let cursor = head;

        while (cursor) {
            let nextTemp = cursor.next; // save next
            cursor.next = prev;         // reverse pointer
            prev = cursor;              // move prev forward
            cursor = nextTemp;          // move curr forward
        }

        return prev; // new head
    }
}

File: Data Structures & Algorithms/delete-n-nodes-after-m-nodes-of-a-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def deleteNodes(self, head: Optional[ListNode], m: int, n: int) -> Optional[ListNode]:
        curr = ListNode(next=head)
        while curr:
            skip = m
            while curr and skip:
                curr = curr.next
                skip -= 1
            
            remove = n
            while curr and curr.next and remove:
                curr.next = curr.next.next
                remove -= 1

        return head